### DISCUSSION TITLE
how to change Thread Pool Size in run time

### DISCUSSION THREAD

**Discussion by aharondavid** [emscripten-core/emscripten#21343] [Q&A]
How can I change the Thread Pool Size in App Initialization.
In the Initialization, I want to determine the Thread Pool Size depending on OS (for example in iOS it will be much smaller than Windows).

**Comment by sbc100** [comment]
The `PTHREAD_POOL_SIZE` setting can accept a JS expression/string.  See https://emscripten.org/docs/porting/pthreads.html#compiling-with-pthreads-enabled where it gives the example of setting it to `navigator.hardwareConcurrency`

**Comment by aharondavid** [comment]
Thanks!
